### DISCUSSION TITLE
Features | 已经安装的 mod 添加标签功能

### DISCUSSION THREAD

**Discussion by 1216892614** [Meloong-Git/PCL#5210] [讨论]
## 使用描述

**打标签:**

1. 打开一个版本游戏的已经安装的 mods 文件夹
2. 选择一个或者多个 mods
3. 点击下方工具栏中的标签功能按钮
4. 在弹窗中选择已有的标签或者输入新创建的标签名称

**根据标签筛选**

1. 打开一个版本游戏的已经安装的 mods 文件夹
2. 选择搜索栏下方的一个或多个标签, 选择的标签会自动排序靠前并展示为选中样式
3. 再次点击已经选择的标签则标签取消选中
4. 结果中标记为已经选中的所有标签的结果靠前显示作为搜索结果, 其他结果置灰靠后显示
5. 如果在此时全选会在选中结果中额外增加搜索结果

## 设计解释

- 搜索栏下方的一个或多个标签
  现有用于展示 `全部(42)` 或者 `搜索结果(16)` 的位置. `搜索结果` 依然可以作为一个标签保留, 但是全部应该撤除. 除了搜索结果, 其他的标签应该可能也需要计数功能, 这是一个可选功能实现. 所有筛选后结果的数量可能需要一个新的计数设计.

  ```
  (搜索结果(12)) (x 标签3(33)) (标签1(4)) (标签2(5)) | 共 45 个结果
  ^              ^             ^                        ^
  |              |             |                        |
  |              |             |                 需要新的搜索结果计数设计
  |              |     未被选中的标签靠后, 点击后被选中
  |      已经选中的标签再次点击可以反选
  存在搜索字符串的时候, 搜索结果也是一个标签, 但是默认选中, 无法反选
  ```

- 选中结果靠前显示, 其他结果靠后置灰显示
  建议搜索结果也改成这样, 这个设计主要是为了让用户明白自己在当前界面的选择不会清除之前界面的选择内容

- 结果中标记为已经选中的所有标签的结果
  求标签的差集, 缩小选择范围而非增大. 如果需要并集则通过多选择几次也能实现, 差集明显是更加符合筛选需求的通用功能.

**Comment by tangge233** [comment]
#277

目前可以通过搜索框搜索有固定标签的模组（科技、服务器、优化、前置等标签）
如果目前对此功能有需求的人很多的话感觉可以考虑下？ @LTCatt

**Comment by LTCatt** [comment]
先 reopen #277，转 #5214
